### PR TITLE
Simplify demo: Remove encryption step for challenge nonce

### PR DESCRIPTION
**Pull request by tomoveu**
Goal of this PR is to enable a simpler demo by eliminating one step from the attestation process:
* Challenge is not encrypted on Backend side
* There is no need to decrypt challenge on Agent side

`Challenge is still part of the TPMS_ATTEST, thus the verification path remains exactly the same.`

@thomas-fossati please let me know if we could add an alternative `api-integration.puml` with the encryption/decryption step presevered, just for completeness, and perhaps name the puml/png files `api-extended-integration.puml`?

### CODE REVIEW

**Comment by tomoveu** [comment]
Only FYI: I had to force push because my main was behind the veraison:main, and had to rebase.

File: docs/sequence-diagrams/api-integration.puml [hunk 1]
```diff
@@ -56,10 +54,8 @@ group periodic evidence submission
 	BE -> V_EviVfy: POST /newSession
 	V_EviVfy -> V_EviVfy: /session/456=>{"nonce"="another-challenge", ...}
 	V_EviVfy -> BE: Status: 201, Location: /session/456, Body: {"nonce"="another-challenge", ...}
-	BE -> BE: Encrypt(EK_pub, "another-challenge") -> challenge-blob
 	BE -> Node: Status: 201, Body: {challenge-blob}
```
Comment (thomas-fossati) [inline docs/sequence-diagrams/api-integration.puml:57]: ditto -- though now it's `"another-challenge"`?

**Review by thomas-fossati — COMMENTED** [review]
thanks Dimi.  I've left a couple of clarification requests inline.

**Review by thomas-fossati — APPROVED** [review]
🚢 it!

> @thomas-fossati please let me know if we could add an alternative api-integration.puml with the encryption/decryption step presevered, just for completeness, and perhaps name the puml/png files api-extended-integration.puml?

Agree it makes sense to keep it around.

**Pull request merged**